Added skip props flag

maslowalex opened the pull request.

jechol committed:
nix for m1 mac

jechol committed:
Merge pull request #46 from jechol/nix-for-m1-mac

nix for m1 mac

jechol committed:
comment for cachix usage.

QuinnWilton committed:
Update .tool-versions

The Elixir version in .tool-versions was below the minimum version
supported by the package.

QuinnWilton committed:
Delete config/config.exs

This resolves a deprecation warning to do with calling `use Mix.Config`
Since libraries aren't intended to set application config anyway,
we should just delete the directory.

QuinnWilton committed:
Update credo: 1.5.5 => 1.6.2

QuinnWilton committed:
Update earmark: 1.4.14 => 1.4.20

QuinnWilton committed:
Update ex_doc: 0.23.0 => 0.26.0

QuinnWilton committed:
Fix alignment of mix.exs

QuinnWilton committed:
Set preferred_cli_env for :quality to :test

By default, if MIX_ENV is unset, mix tasks run using the :dev
environment. This is required to ensure that tests are run when
calling `mix quality` with no explicit environment.

QuinnWilton committed:
Add "espec" to :quality task

QuinnWilton committed:
Remove unused test definition: TypeClassSpec.Applicative.wrap/2

This resolves #42

QuinnWilton committed:
Bump patch version: 1.2.7 => 1.2.8

QuinnWilton committed:
Move config/.credo.exs to .credo.exs

There's no longer any configuration files for this project, and
I don't believe Credo was finding this file as it was.

QuinnWilton committed:
Merge pull request #47 from witchcrafters/quinn/1.2.8

Bump to 1.2.8 + miscellaneous maintenance

jechol committed:
ignore .mix

jechol committed:
Use nixpkgs 21.11

jechol committed:
Remove nix/

jechol committed:
Merge pull request #48 from jechol/main

nixpkgs 21.11 based JIT disabled OTP24, Elixir 1.13 for Mac M1

maslowalex committed:
Skip type_class property check by default

cognivore reviewed (approved):
LGTM! If you have tested the changes, feel free to merge!

maslowalex committed:
Fixed quality

cognivore commented on lib/type_class.ex:
```diff
@@ -252,29 +252,31 @@ defmodule TypeClass do
-          The type class #{unquote(class)} has been forced to bypass \
-          all property checks for all data types. This is very rarely valid, \
-          as all type classes should have properties associted with them.
+          # IO.warn("""
```
I think that this should be another configuration option + ENV option.

maslowalex committed:
Removed config from type_class

cognivore committed:
Problem: libraries with default configs are unpredictable

Solution:

 - Remove the config
 - Move the ENV / config reading into `type_class.exs`
 - Bump the version in mix.exs and establish doma fork

maslowalex committed:
Added another option to show prop check related warnings, edited README

cognivore commented:
https://github.com/witchcrafters/type_class/commit/ddd369949654e413a948be1d4ed1fc4b259f2f7c

Just checking in that you've added logic like this.

maslowalex commented:
> 

Should we add call to `Application` env too? Like with `force_type_class`?

cognivore commented on lib/type_class.ex:
```diff
@@ -141,22 +141,21 @@ defmodule TypeClass do
-            high_prio
-          end
-        end
+				# @doc "Force type classes by default, but accept configuration options based on FORCE_TYPE_CLASS env variable and :doma_type_class > :force_type_class configuration path."
```
Something happened to the indentation here.

maslowalex committed:
Fixed identation

cognivore commented on lib/type_class.ex:
```diff
@@ -263,32 +263,44 @@ defmodule TypeClass do
+      print_warnings = case System.get_env("PRINT_WARNINGS") do
+        nil -> nil
+        "false" -> false
+        _ -> true
```
You have to make sure that your changes agree with each other.

The new default behaviour is to skip compile time tests (FORCE_TYPE_CLASS = true is default).

To match this new behaviour, you have to either:

 - Default PRINT_WARNINGS to false.
 - Rename PRINT_WARNINGS to NOWARN_FORCED_TYPE_CLASS and default it to true (symmetric with the FORCE_TYPE_CLASS).

cognivore commented on lib/type_class.ex:
```diff
@@ -263,32 +263,44 @@ defmodule TypeClass do
-          # For more, please see the TypeClass README:
-          # https://github.com/expede/type_class/blob/master/README.md
-          # """)
+          if print_warnings do
```
Would be cool to have this as a configuration option too, just as the FORCE_TYPE_CLASS is.

cognivore reviewed (changes requested):
- Match the semantics of these changes with the `FORCE_TYPE_CLASS`
 - Add a low_priority configuration option here, analogously with `FORCE_TYPE_CLASS`
 - Fix indentation in the `force_type_class_const` computation.

maslowalex committed:
Match warnings env passign logic with force_type_class

maslowalex committed:
Prepare to release
